internal/dto/registry.go

```diff
@@ -73,22 +85,19 @@ func (registries Registries) Parse(logger log.Logger) (Registries, error) {
 		}
 		username, err := LookupEnvVar(registries[i].Url, username_envvar, logger)
 		if err != nil {
-			logger.Errorf("Push argument detected but error found. Not attempting to build or push"+
-				" to %s from error: (%w)", registries[i].Url, err)
```

Comment:
I don't know if this is conventional Golang or not, but this idiom seems kind of clunky.

Rather than joining the two errors together, I suggest "wrapping" the original one:
```go
type ErrorPushArg struct {
	// location that will receive payload
	target string
	// argument required to upload to target location
	arg string
	// error encounter during push
	err error
}

func (e ErrorPushArg) Error() string {
	return fmt.Sprintf("Push argument %q detected for %q, not attempting to build or push due to error: %w.", e.arg, e.target, e.err)
}
```
This streamlines the uses, like the one here, slightly:
```go
			logger.Errorf("%s", ErrorPushArg{registries[i].Url, "username", err})
```
(Should that be a `%v` or a `%w` rather than a `%s`?)

Reply:
```
The error type is an interface type. An error variable represents any value that can describe itself as a string.
```
[error handling in golang](https://go.dev/blog/error-handling-and-go). In this case, you cannot use the wrapping format directive, `%w`, because the error is not being passed to `fmt.Errorf()`. Out of `%s` and `%v`, it's probably better form to use `%v`.

Reply:
Since the errors are being logged, and not returned through the call stack, then it doesn't matter where you concatenate them into a string, either in the error object's `Error()`, like you did, or at `logger.Errorf(%v %v, ErrorPushArg{}, err)`.

Also, your `Error()` implementation consumes the underlying error as extra information for `ErrorPushArg`, so if we did care about the underlying error's type (in addition to our custom error type) we would not be able to validate it. My rule of thumb is to not consume underlying errors when I implement custom errors, and to return both of them wrapped together, so that you can see both of the error types in the list of wrapped errors.

Reply:
> the error is not being passed to `fmt.Errorf()`.

I see that, now...and, given that, I gotta say, `logger.Errorf()` is a really unfortunately choice of function name.

Or, it's an unfortunate implementation...I wonder how hard it would (have) be(en) to make it support `%w` like `fmt.Errorf()` does....